content/concepts/authentication/content.md

```diff
@@ -8,10 +8,155 @@ title = "Authentication"
   sticky = true
 +++
 
-### Referencing secrets
+Often a scaler will require authentication or secrets and config in order for KEDA to check for events and monitor the event rate.  KEDA provides a few secure patterns to add those secrets to a `ScaledOjbect`.
 
-TODO
+## Referencing secrets and config maps
 
-### TriggerAuthentication
+Some metadata parameters will not allow resolving from a literal value, and will instead require a reference to a secret, config map, or environment variable defined on the target deployment.
 
-TODO
\ No newline at end of file
+For example, if using the [RabbitMQ scaler](https://keda.sh/scalers/rabbitmq-queue/), the `host` parameter may include passwords so is required to be a reference.  You can create a secret with the value of the `host` string, mount that secret to the deployment, and reference it from the `ScaledObject` like below:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {secret-name}
+data:
+  {secret-key-name}: YW1xcDovL3VzZXI6UEFTU1dPUkRAcmFiYml0bXEuZGVmYXVsdC5zdmMuY2x1c3Rlci5sb2NhbDo1Njcy #base64 encoded per secret spec
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {deployment-name}
+  namespace: default
+  labels:
+    app: {deployment-name}
+spec:
+  selector:
+    matchLabels:
+      app: {deployment-name}
+  template:
+    metadata:
+      labels:
+        app: {deployment-name}
+    spec:
+      containers:
+      - name: {deployment-name}
+        image: {container-image}
+        envFrom:
+        - secretRef:
+            name: {secret-name}
+---
+apiVersion: keda.k8s.io/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {deployment-name}
+  namespace: default
+  labels:
+    deploymentName: {deployment-name}
+spec:
+  scaleTargetRef:
+    deploymentName: {deployment-name}
+  triggers:
+  - type: rabbitmq
+    metadata:
+      queueName: hello
+      host: {secret-key-name}
+      queueLength  : '5'
+```
+
+If you have multiple containers in a deployment, you will need to include the name of the container that has the references in the `ScaledObject`.  If you do not include a `containerName` it will default to the first container.  KEDA will attempt to resolve references from secrets, config maps, and environment variables of the container.
+
+> TIP: If creating a deployment yaml that references a secret, be sure the secret is created before the deployment that references it, and the scaledObject after both of them to avoid race conditions.
```

Comment:
```suggestion
> TIP: If creating a deployment yaml that references a secret, be sure the secret is created before the deployment that references it, and the scaledObject after both of them to avoid invalid references.
```